engine/src/main/java/com/datatorrent/stram/engine/GenericNode.java

```diff
@@ -656,6 +744,43 @@ public final void run()
 
   }
 
+  protected void forwardToSinks(boolean delay, Object o)
+  {
+    if (!delay) {
+      for (int s = sinks.length; s-- > 0; ) {
+        sinks[s].put(o);
+      }
+      controlTupleCount++;
+    }
+  }
+
+  /**
+   * Populate {@link #reservoirPortMap} with information on which reservoirs are connected to which input ports
+   */
+  protected void populateReservoirInputPortMap()
+  {
+    for (Entry<String,Operators.PortContextPair<InputPort<?>>> entry : descriptor.inputPorts.entrySet()) {
+      if (entry.getValue().component instanceof InputPort) {
```

Comment:
component will be instance of InputPort by default unless unless component is null.